src/omnicore/tx.cpp

```diff
@@ -165,22 +183,38 @@ bool CMPTransaction::interpret_Transaction()
 /** Version and type */
 bool CMPTransaction::interpret_TransactionType()
 {
-    if (pkt_size < 4) {
-        return false;
+    if (encodingClass == OMNI_CLASS_D) {
+        int i = 0;
+
+        std::vector<uint8_t> vecVersionBytes = GetNextVarIntBytes(i);
```

Comment:
Great catch, you're right...

Do you think this is sufficient to catch all potential calls to `GetNextVarIntBytes()` that may overrun the end of the payload?

```diff
 std::vector<uint8_t> CMPTransaction::GetNextVarIntBytes(int &i) {
     std::vector<uint8_t> vecBytes;
 
+    if (i >= pkt_size) {
+        return vecBytes;
+    }
+
     do {
         vecBytes.push_back(pkt[i]);
         if (!IsMSBSet(&pkt[i])) break;
```